docs/madr/decisions/056-meshexternalservice-and-zoneegress.md

```diff
@@ -0,0 +1,239 @@
+# MeshExternalService policies and ZoneEgress
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/10897
+
+## Context and Problem Statement
+
+Configuration (like timeouts, retries, circuit breakers) of MeshExternalService can be applied both on a sidecar and on an egress.
+
+This MADR will go into the details of each policy type and the placement to figure out what is the best option.
+
+## Decision Drivers
+
+* flexible enough that we don't limit users in scenarios where it makes sense to configuration in both/one place
+* restrictive enough so that it's hard for the user to "shoot themselves in the foot" (e.g. having squared no. of circuit breakers)
+
+## Considered Options
+
+* configurable where it makes sense on both sidecar and egress, predefined where only one makes sense
+* predefined mixed approach on a case-by-case basis
+* only on the egress
+* only on the sidecar
+
+## Decision Outcome
+
+Chosen option: "Configurable where it makes sense on both sidecar and egress, predefined where only one makes sense"
+because it fits the decision drivers (flexibility / restiveness).
+
+## Pros and Cons of the Options
+
+### Configurable where it makes sense on both sidecar and egress, predefined where only one makes sense
+
+#### Table summary
+
+The table below shows a summary of policies and if it makes sense to configure them on:
+- sidecar
+- egress
+- on both
+
+and if it makes sense for the user to pick where.
+
+| Policy                    | On sidecar   | On egress    | On both | User configurable |
+|---------------------------|--------------|--------------|---------|-------------------|
+| MeshAccessLog             | Yes          | Yes          | Yes     | Yes               |
+| MeshCircuitBreaker        | Yes          | Maybe        | No      | No                |
+| MeshFaultInjection        | Maybe        | Yes          | No      | No                |
+| MeshHealthCheck           | Yes          | Maybe        | No      | No                |
+| MeshHealthCheck           | Yes          | Maybe        | No      | No                |
+| MeshMetric                | Yes          | Yes          | Yes     | Maybe             |
+| MeshProxyPatch            | Yes          | Yes          | Yes     | Yes               |
+| MeshRateLimit             | Probably not | Yes          | No      | No                |
+| MeshRetry                 | Yes          | Maybe        | No      | Maybe             |
+| MeshTimeout               | Yes          | Probably not | No      | No                |
+| MeshTrace                 | Yes          | Yes          | Yes     | Maybe             |
+| MeshTrafficPermission     | Maybe        | Yes          | Maybe   | Maybe             |
```

Comment:
MeshTrafficPermission currently works for ExternalService to support AutoreachableServices

Reply:
Good to note this but the table is not about what currently works for ES. Can I resolve this conv?